### ISSUE TITLE
bug(engine): a directory-level tokenizer.htok is applied to every model in a shared directory (regression from #2197)

### ISSUE THREAD

**Issue by bong-water-water-bong** [1bit-MONSTER/1bit-MONSTER#2205]
## Summary

The tokenizer lookup added in #2197 consults **`<model_dir>/tokenizer.htok` first**, then `<stem>.htok`. For a **flat directory holding many models** — the normal case, e.g. `~/models` with ~30 GGUFs — that means **one model's tokenizer is applied to every model in the directory**.

Measured today on strixhalo while running the #1942 hybrid acceptance:

```
Loaded BPE tokenizer from /home/bcloud/models/tokenizer.htok (BOS=0, EOS=106)
```

`~/models/tokenizer.htok` is a **zaya** vocabulary (9,045,287 B, `HTOK` v2 — produced for the zaya acceptance in the previous goal), and it was loaded for a **Qwen3-Coder-30B-A3B** model living in the same directory. The result is garbage *text* output — with correct token-level behaviour underneath — which is exactly the shape that makes a decode look broken when it is not.

## Why it happens

PR #2197 added the artifact-adjacent lookup so a native container could be detokenised without a GGUF sibling. The candidate order is:

1. `<dir>/tokenizer.htok`  ← wins for **every** model in the directory
2. `<stem>.htok`
3. the existing GGUF-sibling logic
4. the weights-dir fallback

The directory-level name is a *single-model* convention (a model directory containing `model.q4nx` + `tokenizer.htok`), but `~/models` is a *shared* directory, so (1) fires before (2) can.

## Impact

- Text output is silently wrong (ids→text mapping) for any model sharing a directory with one that has a `tokenizer.htok`.
- Token-level work (logits/ids, the §5.2 acceptance) is unaffected — which is why it went unnoticed there.

## Proposed fix (small)

Make the per-model candidate win, and treat the directory-level file as a *fallback*:

1. `<stem>.htok` first (per-model), then
2. `<dir>/tokenizer.htok` **only when the directory contains a single model artifact** (checked against the registry, or simply: only if no `<stem>.htok` exists *and* the directory holds one model), then
3. the existing GGUF/weights-dir path.

Alternatively, drop (2) entirely and require per-model naming — simpler, and the GGUF-sibling path already covers the common case.

## Reproduce

```bash
ls -l ~/models/tokenizer.htok      # a zaya vocab, if the previous goal's provisioning is present
~/wt/acc-postr2/build/1bit unified --port 8181 -m ~/models/Qwen3-Coder-30B-A3B-Instruct-Q4_K_M.gguf
# → "Loaded BPE tokenizer from /home/bcloud/models/tokenizer.htok (BOS=0, EOS=106)"
```

Found while verifying #1866/#1942; reported separately because it is a different defect, introduced by #2197.
